sponsorship_switzerland/models/contracts.py

```diff
@@ -413,3 +414,32 @@ def _reconcile_open_amount(self):
             delay = datetime.now() + relativedelta(seconds=15)
             if invoices:
                 invoices.with_delay(eta=delay).group_or_split_reconcile()
+
+    @api.multi
+    @job(default_channel='root.recurring_invoicer')
+    @related_action(action='related_action_contract')
+    def _clean_invoices(self, since_date=None, to_date=None, keep_lines=None):
+        today = datetime.now()
+        inv_lines = self.invoice_line_ids.filtered(lambda r: r.state == 'open'
+                                                   or (r.state == 'paid'
+                                                      and r.date > today))
+
+        for line in inv_lines:
+            line.invoice_id.cancel_payment_lines()
+
+        return super(RecurringContracts, self)._clean_invoices(
+            since_date, to_date, keep_lines)
+
+    @api.model
+    def _needaction_count(self, domain=None):
```

Comment:
This function has nothing to do with your pull request. Please remove this code.

Reply:
Yes, I made a mistake during the merge with devel